### DISCUSSION TITLE
Sunbird ED local installation - googleOauthHelper error.

### DISCUSSION THREAD

**Discussion by Kovendan03** [Sunbird-Ed/Community#204] [Installation]
Hi Team,

I Followed the documentation - https://github.com/Sunbird-Ed/SunbirdEd-portal, for setting up a local installation of ED in ubuntu server. But unfortunately, when i run "npm run server" in src/app folder, i was getting the below error.

{"message":{"msg":"logger initialized with LEVEL= debug"},"level":"info","channel":"sunbird","env":"local.sunbird.portal","pdata":{"id":"local.sunbird.portal","ver":"4.10.0","pid":"sunbird-portal-backend"},"timestamp":"2022-09-29 20:42:54"}
express-session deprecated req.secret; provide secret option server.js:106:3
{"error":{"message":"Cannot find module './jwtHelper'\nRequire stack:\n- /root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js\n- /root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js\n- /root/SunbirdEd-portal/src/app/server.js","stack":"Error: Cannot find module './jwtHelper'\nRequire stack:\n- /root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js\n- /root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js\n- /root/SunbirdEd-portal/src/app/server.js\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)\n    at Function.Module._load (node:internal/modules/cjs/loader:804:27)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js:11:23)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:839:12)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js:3:82)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:839:12)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/server.js:241:1)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)","code":"MODULE_NOT_FOUND","requireStack":["/root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js","/root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js","/root/SunbirdEd-portal/src/app/server.js"]},"level":"error","message":"uncaughtException: Cannot find module './jwtHelper'\nRequire stack:\n- /root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js\n- /root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js\n- /root/SunbirdEd-portal/src/app/server.js\nError: Cannot find module './jwtHelper'\nRequire stack:\n- /root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js\n- /root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js\n- /root/SunbirdEd-portal/src/app/server.js\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)\n    at Function.Module._load (node:internal/modules/cjs/loader:804:27)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js:11:23)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:839:12)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js:3:82)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:839:12)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/server.js:241:1)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)","stack":"Error: Cannot find module './jwtHelper'\nRequire stack:\n- /root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js\n- /root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js\n- /root/SunbirdEd-portal/src/app/server.js\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)\n    at Function.Module._load (node:internal/modules/cjs/loader:804:27)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js:11:23)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:839:12)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js:3:82)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:839:12)\n    at Module.require (node:internal/modules/cjs/loader:1028:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/root/SunbirdEd-portal/src/app/server.js:241:1)\n    at Module._compile (node:internal/modules/cjs/loader:1126:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)\n    at Module.load (node:internal/modules/cjs/loader:1004:32)","exception":true,"date":"Thu Sep 29 2022 20:42:54 GMT+0530 (India Standard Time)","process":{"pid":2957835,"uid":0,"gid":0,"cwd":"/root/SunbirdEd-portal/src/app","execPath":"/usr/bin/node","version":"v16.17.0","argv":["/usr/bin/node","/root/SunbirdEd-portal/src/app/server"],"memoryUsage":{"rss":185958400,"heapTotal":128815104,"heapUsed":98544544,"external":15603681,"arrayBuffers":1967497}},"os":{"loadavg":[1.91,1.67,2.04],"uptime":103367.99},"trace":[{"column":15,"file":"node:internal/modules/cjs/loader","function":"Module._resolveFilename","line":956,"method":"_resolveFilename","native":false},{"column":27,"file":"node:internal/modules/cjs/loader","function":"Module._load","line":804,"method":"_load","native":false},{"column":19,"file":"node:internal/modules/cjs/loader","function":"Module.require","line":1028,"method":"require","native":false},{"column":18,"file":"node:internal/modules/cjs/helpers","function":"require","line":102,"method":null,"native":false},{"column":23,"file":"/root/SunbirdEd-portal/src/app/helpers/googleOauthHelper.js","function":null,"line":11,"method":null,"native":false},{"column":14,"file":"node:internal/modules/cjs/loader","function":"Module._compile","line":1126,"method":"_compile","native":false},{"column":10,"file":"node:internal/modules/cjs/loader","function":"Module._extensions..js","line":1180,"method":".js","native":false},{"column":32,"file":"node:internal/modules/cjs/loader","function":"Module.load","line":1004,"method":"load","native":false},{"column":12,"file":"node:internal/modules/cjs/loader","function":"Module._load","line":839,"method":"_load","native":false},{"column":19,"file":"node:internal/modules/cjs/loader","function":"Module.require","line":1028,"method":"require","native":false},{"column":18,"file":"node:internal/modules/cjs/helpers","function":"require","line":102,"method":null,"native":false},{"column":82,"file":"/root/SunbirdEd-portal/src/app/routes/googleSignInRoutes.js","function":null,"line":3,"method":null,"native":false},{"column":14,"file":"node:internal/modules/cjs/loader","function":"Module._compile","line":1126,"method":"_compile","native":false},{"column":10,"file":"node:internal/modules/cjs/loader","function":"Module._extensions..js","line":1180,"method":".js","native":false},{"column":32,"file":"node:internal/modules/cjs/loader","function":"Module.load","line":1004,"method":"load","native":false},{"column":12,"file":"node:internal/modules/cjs/loader","function":"Module._load","line":839,"method":"_load","native":false},{"column":19,"file":"node:internal/modules/cjs/loader","function":"Module.require","line":1028,"method":"require","native":false},{"column":18,"file":"node:internal/modules/cjs/helpers","function":"require","line":102,"method":null,"native":false},{"column":1,"file":"/root/SunbirdEd-portal/src/app/server.js","function":null,"line":241,"method":null,"native":false},{"column":14,"file":"node:internal/modules/cjs/loader","function":"Module._compile","line":1126,"method":"_compile","native":false},{"column":10,"file":"node:internal/modules/cjs/loader","function":"Module._extensions..js","line":1180,"method":".js","native":false},{"column":32,"file":"node:internal/modules/cjs/loader","function":"Module.load","line":1004,"method":"load","native":false}],"channel":"sunbird","env":"local.sunbird.portal","pdata":{"id":"local.sunbird.portal","ver":"4.10.0","pid":"sunbird-portal-backend"},"timestamp":"2022-09-29 20:42:54"}

I did configured the azure related settings in the "environmentVariablesHelper.js" but still getting the above error. referred  the discussion in the same community - https://github.com/Sunbird-Ed/Community/discussions/63.

can you please provide the jwtHelper file to proceed further.

**Comment by gandham-santhosh** [comment]
@rajeevsathish @sharathkashyap can you please help here?

**Reply by rajeshkumaravel** [reply]
@Kovendan03 
Please share more details on 
|              Key             | Version / Values |
|:----------------------------:|:--------------:|
|              Operating System              |                |
|            SunbirdEd Version            |                |
|              Angular CLI             |                |
|            NodeJS            |                |
|              npm             |                |
| System environment variables |                |

**System environment variables**
|           Key           | Version Values |
|:-----------------------:|:--------------:|
|   sunbird_environment   |                |
|     sunbird_instance    |                |
| sunbird_default_channel |                |
|  sunbird_default_tenant |                |

CC: @sharathkashyap @rajeevsathish

**Reply by luckynvdu** [reply]
@Kovendan03 if the issue resolved?

**Comment by sharathkashyap** [comment]
@Kovendan03 We may have to close the thread due to inactivity as per community guidelines.
